stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/selectedmessage/SelectedMessageMenu.kt

```diff
@@ -0,0 +1,137 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.Reaction
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.common.state.MessageAction
+import io.getstream.chat.android.common.state.React
+import io.getstream.chat.android.compose.handlers.SystemBackPressedHandler
+import io.getstream.chat.android.compose.state.messages.list.MessageOptionState
+import io.getstream.chat.android.compose.ui.components.messageoptions.MessageOptions
+import io.getstream.chat.android.compose.ui.components.messageoptions.defaultMessageOptionsState
+import io.getstream.chat.android.compose.ui.components.reactionoptions.ReactionOptions
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+
+/**
+ * A Composable function that turns [SelectedMessageOptions] into a menu by wrapping it inside of [Surface].
+ *
+ * @param message Used to gather the necessary message information (such as ownership, id, etc.) in order to properly display [SelectedMessageOptions].
+ * @param modifier Compose UI [Modifier] that is applied to the internally used [Surface].
+ * @param shape The [Shape] applied to the internally used [Surface].
+ * @param overlayColor The color applied to the internally used [Box] that wraps [Surface].
+ * @param reactionTypes By default used to display all the available reaction options inside of [headerContent].
+ * @param messageOptions By default used to display all available message options inside of [bodyContent].
+ * @param onMessageAction By default a Handler used to propagate click events on individual reaction and message option elements
+ * inside of the header and body content.
+ * @param onDismiss Handler called when the menu is dismissed.
+ * @param headerContent Leading vertical Composable slot.
+ * @param bodyContent Trailing vertical Composable slot.
+ */
+@Composable
+public fun SelectedMessageMenu(
+    message: Message,
+    modifier: Modifier = Modifier,
+    shape: Shape = ChatTheme.shapes.bottomSheet,
+    overlayColor: Color = ChatTheme.colors.overlay,
+    reactionTypes: Map<String, Int> = ChatTheme.reactionTypes,
+    messageOptions: List<MessageOptionState>,
+    onMessageAction: (MessageAction) -> Unit,
+    onDismiss: () -> Unit = {},
+    headerContent: @Composable ColumnScope.() -> Unit = {
+        ReactionOptions(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp, vertical = 8.dp),
+            reactionTypes = reactionTypes,
+            onReactionOptionClicked = {
+                onMessageAction(
+                    React(
+                        Reaction(
```

Comment:
Can we make this into one line or something? looks pretty weird broken up completely

Reply:
Reformatted to:

```kotlin
                   React(
                        reaction = Reaction(messageId = message.id, type = it.type),
                        message = message
                    )
```